cmd/osbuild-worker/jobimpl-depsolve.go

```diff
@@ -11,12 +11,18 @@ type DepsolveJobImpl struct {
 	RPMMDCache string
 }
 
-func (impl *DepsolveJobImpl) depsolve(packageSets map[string]rpmmd.PackageSet, repos []rpmmd.RepoConfig, modulePlatformID, arch, releasever string) (map[string][]rpmmd.PackageSpec, error) {
+func (impl *DepsolveJobImpl) depsolve(packageSets map[string]rpmmd.PackageSet, repos []rpmmd.RepoConfig, modulePlatformID, arch, releasever string, packageSetsRepositories map[string][]rpmmd.RepoConfig) (map[string][]rpmmd.PackageSpec, error) {
 	rpmMD := rpmmd.NewRPMMD(impl.RPMMDCache, "/usr/libexec/osbuild-composer/dnf-json")
 
 	packageSpecs := make(map[string][]rpmmd.PackageSpec)
 	for name, packageSet := range packageSets {
-		packageSpec, _, err := rpmMD.Depsolve(packageSet, repos, modulePlatformID, arch, releasever)
+		repositories := repos
+		if packageSetRepositories, ok := packageSetsRepositories[name]; ok {
+			repositories = make([]rpmmd.RepoConfig, 0, len(repos)+len(packageSetRepositories))
+			repositories = append(repositories, repos...)
+			repositories = append(repositories, packageSetRepositories...)
+		}
```

Comment:
I think moving the `make` call here would make it a bit more readable:

```
repositories := make([]rpmmd.RepoConfig, len(repos))
copy(repositories, repos)
if packageSetRepositories, ok := packageSetsRepositories[name]; ok {
	repositories = append(repositories, packageSetsRepositories[name])
}
```

Reply:
My intention here was to make a copy of repos and allocate a new map only when it's needed, as on 90% of cases we may not have custom repos. what do you think?

Reply:
it's true that the usual path is without custom repos, then again this won't have any significant performance impact either way. It's not very important either way.